nmdc_automation/re_iding/base.py

```diff
@@ -81,19 +97,8 @@ def __init__(self, api_client: NmdcRuntimeApi, data_dir: str,
         with open(template_file, "r") as f:
             self.workflow_template = yaml.safe_load(f)["Workflows"]
         # collector to track ID changes as (type, old_id, new_id)
-        self.id_changes = []
-
-    def write_id_changes(self, study_id: str):
-        """
-        Write ID changes to a file.
-        """
-        filename = f"{study_id}_id_changes.tsv"
-        with open(filename, "w") as f:
-            writer = csv.writer(f, delimiter="\t")
-            writer.writerow(["type", "old_id", "new_id"])
-            for change in self.id_changes:
-                writer.writerow(change)
-
+        self.updated_record_identifiers = set()
```

Comment:
I think one advantage of writing to a file during the course of the script is that, if the script were to crash for some reason, there would at least be a file that people could look at in order to determine what IDs had already been changed.

Since we will be doing the re-ID-ing via an ETL process, if the script crashes for some reason and we don't know which IDs have already been changed, we can just start over from the production dump.